fex/7z_C/7zAlloc.c

```diff
@@ -20,6 +20,7 @@ int g_allocCountTemp = 0;
 
 void* SzAlloc(void* p, size_t size)
 {
+    p = p;
```

Comment:
Which warning does this fix? Is there a way to do this differently? We can also just turn off this warning too. Because this is kind of stupid.

Reply:
Stuff like this:
```
/home/denisfa/Git/visualboyadvance-m/fex/7z_C/7zAlloc.c: In function ‘SzAlloc’:
/home/denisfa/Git/visualboyadvance-m/fex/7z_C/7zAlloc.c:21:21: warning: unused parameter ‘p’ [-Wunused-parameter]
 void* SzAlloc(void* p, size_t size)
```

We can supress this by using `__attribute__((unused))` (works for gcc and clang):
```
void* SzAlloc(void* p __attribute__((unused)), size_t size)
```

My question here is why do we have the `void *p` param here, since it is not used at all. We are coding this function C-style, so it is not like a virtual function declaration somewhere demanding this from us.